server/webhook.go

```diff
@@ -429,6 +429,11 @@ func (s *Server) filterRepository(ctx context.Context, webhookEndpointID string,
 }
 
 func (*Server) isWebhookEventBranch(pushEventRef, branchFilter string) (bool, error) {
+	// TODO(d): we will accept all pushes from wildcard branch filter for now.
+	// We should implement our own branch filter later.
+	if strings.HasSuffix(branchFilter, "*") {
```

Comment:
This is too general, GitLab user may specify "feature-*" as the branch filter.

They definitely don't want to match branch "foobar"

Reply:
+1

Reply:
This is a quick fix that we need to tightening this part later.